modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc

```diff
@@ -303,17 +304,20 @@ CONCAT_WS(''," . $DB->quote($dir_path) . ",'trashbin/',SUBSTRING_INDEX(minc_loca
             'minYear'         => $study['startYear'] - $study['ageMax'],
             'maxYear'         => $study['endYear'] - $study['ageMin']
         );
-        $problemOptions = array(
-            '' => 'Any',
-            'CandID and PSCID do not match database'
-            => 'Candidate Mismatch',
-            'Could not identify scan type'
-            => 'Could not identify scan type',
-            'Protocol Violation'
-            => 'MRI Protocol Check violation',
-            'Visit label does not exist'
-            => 'Visit label does not exist'
+
+        $problemOptions = ['' => 'Any'];
+
+        // Build a list of existing problem types based on data in db
+        $problemTypes = $db->pselect(
+            "SELECT DISTINCT MRICandidateErrors.Reason FROM MRICandidateErrors
+            UNION 
+            SELECT DISTINCT 'Could not identify scan type' FROM mri_protocol_violated_scans
```

Comment:
`DISTINCT` and  `From table name` should be removed in the second and third select statements